sdks/python/apache_beam/transforms/util.py

```diff
@@ -485,33 +485,57 @@ class ReshufflePerKey(PTransform):
   """
 
   def expand(self, pcoll):
-    class ReifyTimestamps(DoFn):
-      def process(self, element, timestamp=DoFn.TimestampParam):
-        yield element[0], TimestampedValue(element[1], timestamp)
+    windowing_saved = pcoll.windowing
+    if windowing_saved.is_default():
+      # In this (common) case we can use a trivial trigger driver
+      # and avoid the (expensive) window param.
+      _globally_windowed = window.GlobalWindows.windowed_value(None)
```

Comment:
Style: why does this variable get a `_` prefix?

Reply:
Removed.